doc/api/corepack.md

```diff
@@ -0,0 +1,96 @@
+# Corepack
+
+<!-- introduced_in=REPLACEME -->
+<!-- type=misc -->
+
+> Stability: 1 - Experimental
+
+_[Corepack][]_ is an experimental tool shipped with Node.js to help with
+managing versions of your package managers. It exposes binary proxies for
+[Yarn][] and [pnpm][] that, when called, locate the package managers configured
+for your local projects, transparently install them, and finally run them
+without requiring explicit user interaction.
+
+This feature allows you to make sure that everyone in your team is using
+exactly the package manager version you want them to without having to check
+them into your repository or managing complex migrations across developers'
+systems.
+
+## Workflows
+
+### Enabling the feature
+
+Due to its experimental status Corepack currently needs to be explicitly
+enabled to have any effect. To do that, first make sure that neither Yarn nor
+pnpm are installed (uninstall them if needed), then run:
+
+```bash
+corepack enable
+```
+
+From this point forward, any call to the `yarn`, `pnpm`, or `pnpx` binaries
+will work without further setup. Should you experience a problem, just run
+`corepack disable` to remove the proxies from your system (and consider opening
+up an issue on the [Corepack repository][] to let us know).
+
+### Configuring a project
+
+The Corepack proxies will find the closest `package.json` files in your
+directory hierarchy to extract their `packageManager` property, formatted as
+such:
+
+```json
+{
+  "packageManager": "yarn@1.22.10"
+}
+```
+
+If the requested package manager is supported (only Yarn and pnpm are at the
+moment), Corepack will make sure that all calls to the relevant binaries (in
+this case any command starting with `yarn`) are ran against the specified
+version.
```

Comment:
I think this should be documented in `packages.md`, in the `package.json` field definitions section:

https://github.com/nodejs/node/blob/8e3e46d528ba3bf41413f7a24f9eeaa796742c70/doc/api/packages.md#L781-L787